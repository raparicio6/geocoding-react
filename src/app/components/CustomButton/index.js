/* eslint-disable react/button-has-type */
import React from 'react';
import { CircularProgress } from '@material-ui/core';
import { string, bool } from 'prop-types';

import styles from './styles.module.scss';

function CustomButton({ label, className, type, loading, disabled, ...props }) {
  return (
    <button className={`${styles.button} ${className}`} type={type} disabled={disabled || loading} {...props}>
      {loading ? (
        <div className="row center middle">
          <div style={{ opacity: 0 }}>{label}</div>
          <div style={{ position: 'absolute' }}>
            <CircularProgress className={styles.spinner} />
          </div>
        </div>
      ) : (
        label
      )}
    </button>
  );
}

CustomButton.propTypes = {
  label: string.isRequired,
  className: string,
  disabled: bool,
  loading: bool,
  type: string
};

CustomButton.defaultProps = {
  className: '',
  disabled: false,
  loading: false,
  type: 'button'
};

export default CustomButton;
