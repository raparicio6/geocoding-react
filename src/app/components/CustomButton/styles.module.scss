@import '../../../scss/colors';
@import '../../../scss/variables';

.button {
  background-color: $royal-blue;
  border-color: $white;
  border-radius: 4px;
  border-style: solid;
  border-width: 2px;
  color: $white;
  font-family: Texta-Medium;
  font-size: 26px;
  font-weight: 700;
  max-height: 55px;
  padding: 20px 16px 16px;
  transition-duration: 0.25s;

  &:disabled {
    background: $concrete;
    color: $very-light-grey;
    cursor: not-allowed;
  }

  &:not([disabled]):not(.disabled):hover { // stylelint-disable-line selector-max-attribute
    transform: translateY(-4px);
  }
}

.spinner {
  height: 30px !important; // stylelint-disable-line declaration-no-important
  width: 30px !important; // stylelint-disable-line declaration-no-important
}

@media (max-width: $breakpoint-tablet) {
  .button {
    font-size: 22px;
    padding-top: 14px;
    padding-bottom: 10px;
  }

  .spinner {
    height: 20px !important; // stylelint-disable-line declaration-no-important
    width: 20px !important; // stylelint-disable-line declaration-no-important
  }
}

@media (max-width: $breakpoint-mobile) {
  .button {
    font-size: 16px;
    padding-top: 10px;
    padding-bottom: 8px;
  }

  .spinner {
    height: 20px !important; // stylelint-disable-line declaration-no-important
    width: 20px !important; // stylelint-disable-line declaration-no-important
  }
}
