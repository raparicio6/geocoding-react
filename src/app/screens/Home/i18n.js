import i18next from 'i18next';

i18next.addResources('en', 'Home', {
  addressesButtonLabel: 'Use Addresses',
  locationButtonLabel: 'Use Locations',
  title: 'Find distances around the world'
});
