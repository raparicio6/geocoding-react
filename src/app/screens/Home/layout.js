import React from 'react';
import { Button } from 'antd';
import './styles.css';

function Home() {
  return <Button type="primary">Button</Button>;
}

export default Home;
