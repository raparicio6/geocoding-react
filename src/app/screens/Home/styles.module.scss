@import '../../../scss/colors';

.background {
  background-color: $wild-sand;
  background-image: url('world-map.png');
  background-position: center;
  background-repeat: no-repeat;
  background-size: 100%;
  min-height: 100vh;
}

.container {
  margin: auto;
}

.title {
  color: $emperor;
  font-family: Texta-Medium;
  font-size: 40px;
  font-weight: 600;
}
