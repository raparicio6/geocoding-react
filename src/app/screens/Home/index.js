import React, { useCallback } from 'react';
import { t } from 'i18next';
import { useHistory } from 'react-router-dom';
import { useDispatch } from 'react-redux';

import CustomButton from '../../components/CustomButton';
import { Routes } from '../../constants';
import TrackingRedux from '../../../redux/TrackingRedux';

function Home() {
  const dispatch = useDispatch();

  const addClick = useCallback(url => dispatch(TrackingRedux.addClick(url)), [dispatch]);

  const history = useHistory();

  const handleAddressesOnClick = useCallback(() => history.push(Routes.ADDRESSES), [history]);
  const handleLocationsOnClick = useCallback(() => history.push(Routes.LOCATIONS), [history]);

  return (
    <div
      onClick={() => {
        addClick(Routes.HOME);
      }}
    >
      <h1 className="title">{t('Home:title')}</h1>
      <div className="row space-around m-top-8">
        <CustomButton label={t('Home:addressesButtonLabel')} onClick={handleAddressesOnClick} />
        <CustomButton label={t('Home:locationButtonLabel')} onClick={handleLocationsOnClick} />
      </div>
    </div>
  );
}

export default Home;
