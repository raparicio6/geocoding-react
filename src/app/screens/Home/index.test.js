import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import CustomButton from '../../components/CustomButton';

import Home from './index';

configure({ adapter: new Adapter() });

describe('Home', () => {
  it('should render', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.exists()).toBeTruthy();
  });
  it('has two CustomButton components', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.find(CustomButton)).toHaveLength(2);
  });
  it('matches the snapshot', () => {
    const tree = toJson(mount(<Home />));
    expect(tree).toMatchSnapshot();
  });
});
