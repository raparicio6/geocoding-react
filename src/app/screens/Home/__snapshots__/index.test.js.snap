// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<Home>
  <h1
    className="title"
  />
  <div
    className="row space-around m-top-8"
  >
    <CustomButton
      className=""
      disabled={false}
      label=""
      loading={false}
      onClick={[Function]}
      type="button"
    >
      <button
        className="button "
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </CustomButton>
    <CustomButton
      className=""
      disabled={false}
      label=""
      loading={false}
      onClick={[Function]}
      type="button"
    >
      <button
        className="button "
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </CustomButton>
  </div>
</Home>
`;
